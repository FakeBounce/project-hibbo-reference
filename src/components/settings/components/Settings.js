import React, { PureComponent } from 'react';
import { Dimensions, View, Text, Image, TouchableOpacity } from 'react-native';
import { string, func, shape } from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet';
import { bind } from 'decko';
import { getTranslations } from 'utils/i18n';

import Carousel from 'shared/components/Carousel';
import FlexibleHeaderLayout from 'shared/components/FlexibleHeaderLayout';

import parent from 'assets/settings/mock/parent.png';
import bigPiggy from 'assets/settings/big-piggy.png';
import cb from 'assets/settings/cb.png';
import colissimo from 'assets/settings/colissimo.png';

const { width } = Dimensions.get('window');

const styles = EStyleSheet.create({
  title: {
    fontFamily: '$fonts.circularBook',
    fontSize: 18,
    color: '$colors.primary',
    textAlign: 'center',
  },
  section: {
    borderBottomWidth: 1,
    borderBottomColor: '$colors.borderGrey',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 40,
    paddingBottom: 40,
  },
  sectionTextWrapper: {
    marginBottom: 40,
  },
  sectionText: {
    fontFamily: '$fonts.circularBook',
    fontSize: 18,
    color: '$colors.lightGrey',
    textAlign: 'center',
  },
  slideWrapper: {
    alignItems: 'center',
  },
  thumbnailItemWrapper: {
    borderRadius: 16,
    borderColor: '$colors.lightGreyBullet',
    padding: 2,
    margin: 5,
  },
  activeThumbnailItemWrapper: {
    borderWidth: 1,
  },
  thumbnail: {
    width: 16,
    height: 16,
    borderRadius: 8,
    backgroundColor: '$colors.lightGreyBullet',
  },
});

const piggyCarousel = [{ uri: bigPiggy }, { uri: bigPiggy }, { uri: bigPiggy }];

const cbCarousel = [{ uri: cb }, { uri: cb }, { uri: cb }];

const renderActiveThumbnail = (item, index) => (
  <View
    key={index}
    style={[styles.thumbnailItemWrapper, styles.activeThumbnailItemWrapper]}
  >
    <View style={styles.thumbnail} />
  </View>
);

const renderEmptyThumbnail = (item, index) => (
  <View key={index} style={styles.thumbnailItemWrapper}>
    <View style={styles.thumbnail} />
  </View>
);

class Settings extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      purchases: [
        {
          id: 1,
        },
        {
          id: 2,
        },
        {
          id: 3,
        },
      ],
      scrollPosY: 0,
    };
  }

  @bind
  onPressBack() {
    this.props.navigation.goBack(null);
  }

  @bind
  onPressParentProfile() {
    this.props.navigation.navigate('ParentProfileForm');
  }

  @bind
  onPressChildProfile() {
    this.props.navigation.navigate('ChildProfileForm');
  }

  @bind
  onPressPaymentMeans() {
    this.props.navigation.navigate('PaymentMeansForm');
  }

  @bind
  onPressPurchase() {
    this.props.navigation.navigate('PurchasesList');
  }

  @bind
  onScroll(event) {
    this.setState({
      scrollPosY: event.nativeEvent.contentOffset.y,
    });
  }

  @bind
  renderPurchaseText() {
    switch (this.state.purchases.length) {
      case 0:
        return (
          <Text style={styles.sectionText}>
            {getTranslations('settings.summary.noOngoingPurchases')}
          </Text>
        );
      case 1:
        return (
          <Text style={styles.sectionText}>
            {getTranslations('settings.summary.oneOngoingPurchase')}
          </Text>
        );
      default:
        return (
          <Text style={styles.sectionText}>
            {getTranslations('settings.summary.multipleOngoingPurchases', {
              number: this.state.purchases.length,
            })}
          </Text>
        );
    }
  }

  @bind
  renderPaymentMeansItem(item, index) {
    return (
      <View key={index} style={[styles.slideWrapper, { width }]}>
        <TouchableOpacity onPress={this.onPressPaymentMeans}>
          <Image source={item.uri} />
        </TouchableOpacity>
      </View>
    );
  }

  @bind
  renderPiggyItem(item, index) {
    return (
      <View key={index} style={[styles.slideWrapper, { width }]}>
        <TouchableOpacity onPress={this.onPressChildProfile}>
          <Image source={item.uri} />
        </TouchableOpacity>
      </View>
    );
  }

  render() {
    return (
      <FlexibleHeaderLayout
        navigation={this.props.navigation}
        title="Settings"
        onBack={this.onPressBack}
        onScroll={this.onScroll}
        scrollPosY={this.state.scrollPosY}
      >
        <View style={styles.section}>
          <TouchableOpacity onPress={this.onPressParentProfile}>
            <View style={styles.sectionTextWrapper}>
              <Text style={styles.title}>
                {getTranslations('settings.summary.myProfile')}
              </Text>
              <Text style={styles.sectionText}>Antoine</Text>
            </View>
            <Image source={parent} />
          </TouchableOpacity>
        </View>

        <View style={styles.section}>
          <View style={styles.sectionTextWrapper}>
            <Text style={styles.title}>
              {getTranslations('settings.summary.myPiggyBanks')}
            </Text>
            <Text style={styles.sectionText}>Marianne</Text>
          </View>
          <Carousel
            items={piggyCarousel}
            renderItem={this.renderPiggyItem}
            renderActiveThumbnail={renderActiveThumbnail}
            renderEmptyThumbnail={renderEmptyThumbnail}
          />
        </View>

        <View style={styles.section}>
          <View style={styles.sectionTextWrapper}>
            <Text style={styles.title}>
              {getTranslations('settings.summary.myPaymentMeans')}
            </Text>
            <Text style={styles.sectionText}>Visa</Text>
          </View>
          <Carousel
            items={cbCarousel}
            renderItem={this.renderPaymentMeansItem}
            renderActiveThumbnail={renderActiveThumbnail}
            renderEmptyThumbnail={renderEmptyThumbnail}
          />
        </View>

        <View style={styles.section}>
          <View style={styles.sectionTextWrapper}>
            <Text style={styles.title}>
              {getTranslations('settings.summary.myPurchases')}
            </Text>
            {this.renderPurchaseText()}
          </View>
          <TouchableOpacity onPress={this.onPressPurchase}>
            <Image source={colissimo} />
          </TouchableOpacity>
        </View>
      </FlexibleHeaderLayout>
    );
  }
}

Settings.propTypes = {
  navigation: shape({
    dispatch: func.isRequired,
    goBack: func.isRequired,
    navigate: func.isRequired,
    setParams: func.isRequired,
    state: shape({
      key: string.isRequired,
      routeName: string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default Settings;
