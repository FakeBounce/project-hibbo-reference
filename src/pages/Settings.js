import { createNavigator, StackRouter } from 'react-navigation';
import RouterTransitioner from '../routes/navigator/RouterTransitioner';

import SettingsComponent from '../components/settings';
import ParentProfileForm from '../components/settings/components/ParentProfileForm';
import ChildProfileForm from '../components/settings/components/ChildProfileForm';
import ChildProfileTransfer from '../components/settings/components/ChildProfileTransfer';
import PaymentMeansForm from '../components/settings/components/PaymentMeansForm';
import PurchasesList from '../components/settings/components/PurchasesList';

const SettingsStack = createNavigator(
  StackRouter({
    SettingsHome: {
      screen: SettingsComponent,
    },
    ParentProfileForm: {
      screen: ParentProfileForm,
    },
    ChildProfileForm: {
      screen: ChildProfileForm,
    },
    ChildProfileTransfer: {
      screen: ChildProfileTransfer,
    },
    PaymentMeansForm: {
      screen: PaymentMeansForm,
    },
    PurchasesList: {
      screen: PurchasesList,
    },
  }),
)(RouterTransitioner);

export default SettingsStack;
