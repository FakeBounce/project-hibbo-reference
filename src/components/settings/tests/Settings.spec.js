import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Settings from '../components/Settings';

describe('Settings component', () => {
  it('should render the component', () => {
    const nav = {
      dispatch: () => {},
      goBack: () => {},
      navigate: () => {},
      setParams: () => {},
      state: {
        key: '',
        routeName: '',
      },
    };
    const value = renderer.create(<Settings navigation={nav} />).toJSON();

    expect(value).toMatchSnapshot();
  });
});
