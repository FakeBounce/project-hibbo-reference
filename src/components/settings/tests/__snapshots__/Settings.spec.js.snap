// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should render the component 1`] = `
<View
  style={
    Array [
      undefined,
      Object {},
    ]
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "height": 90,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      Settings
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={undefined}
    onScroll={[Function]}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={undefined}
            >
              My profile
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={undefined}
            >
              Antoine
            </Text>
          </View>
          <Image
            source={1}
          />
        </View>
      </View>
      <View
        style={undefined}
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            My piggy banks
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            Marianne
          </Text>
        </View>
        <View
          style={undefined}
        >
          <RCTScrollView
            decelerationRate={0}
            horizontal={true}
            onScroll={[Function]}
            scrollEventThrottle={3}
            showsHorizontalScrollIndicator={false}
            snapToAlignment="start"
            snapToInterval={750}
            style={undefined}
          >
            <View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={undefined}
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                style={undefined}
              />
            </View>
            <View
              style={undefined}
            >
              <View
                style={undefined}
              />
            </View>
            <View
              style={undefined}
            >
              <View
                style={undefined}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={undefined}
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            My payment means
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            Visa
          </Text>
        </View>
        <View
          style={undefined}
        >
          <RCTScrollView
            decelerationRate={0}
            horizontal={true}
            onScroll={[Function]}
            scrollEventThrottle={3}
            showsHorizontalScrollIndicator={false}
            snapToAlignment="start"
            snapToInterval={750}
            style={undefined}
          >
            <View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={undefined}
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                style={undefined}
              />
            </View>
            <View
              style={undefined}
            >
              <View
                style={undefined}
              />
            </View>
            <View
              style={undefined}
            >
              <View
                style={undefined}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={undefined}
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            My orders
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            3 purchases ongoing
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
